aesm-client/src/imp/windows.rs

```diff
@@ -323,69 +124,54 @@ impl AesmClient {
         spid: Vec<u8>,
         sig_rl: Vec<u8>,
     ) -> Result<QuoteResult> {
-        use sgx_isa::Report;
         let nonce = [0u8; 64];
         let quote_buffer_size = session.quote_buffer_size(&sig_rl);
         let mut qe_report: Vec<u8> = vec![0; Report::UNPADDED_SIZE];
         let mut quote: Vec<u8> = vec![0; quote_buffer_size as usize];
-        let mut error: AesmError = 0;
 
         unsafe {
-            let interface = self.create_instance()?;
-            if let Some(get_quote) = (*(*interface.inner).vtbl).get_quote {
-                get_quote(
-                    interface.inner,
-                    report.as_ptr(),
-                    report.len() as _,
+            let lib = &self.library;
+
+            let (sig_rl_in, sig_rl_size_in )= match sig_rl.len() {
+                0 => (std::ptr::null(), 0),
+                s => (sig_rl.as_ptr(), s)
+            };
+
+            let error = (lib.get_quote)(
```

Comment:
You should assert that `report` and `spid` are of the appropriate size before passing them to C